domain-management/src/main/java/org/jboss/as/domain/management/notification/registrar/NotificationRegistrarResourceDefinition.java

```diff
@@ -0,0 +1,189 @@
+    public static final SimpleAttributeDefinition MODULE = new SimpleAttributeDefinitionBuilder(ModelDescriptionConstants.MODULE, ModelType.STRING)
+            .setValidator(new StringLengthValidator(1))
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .build();
+
+
+    public static final SimpleAttributeDefinition CODE = new SimpleAttributeDefinitionBuilder(ModelDescriptionConstants.CODE, ModelType.STRING)
+            .setValidator(new StringLengthValidator(1))
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .build();
+
+    public static final PropertiesAttributeDefinition PROPERTIES = new PropertiesAttributeDefinition.Builder(ModelDescriptionConstants.PROPERTIES, true)
+            .setElementValidator(new StringLengthValidator(1, true))
+            .setFlags(AttributeAccess.Flag.RESTART_RESOURCE_SERVICES)
+            .setAllowExpression(true)
+            .build();
+
+
+    private static final AttributeDefinition[] ATTRIBUTES = new AttributeDefinition[] {MODULE, CODE, PROPERTIES};
+
+    static final NotificationRegistrarResourceDefinition INSTANCE = new NotificationRegistrarResourceDefinition();
+
+    private NotificationRegistrarResourceDefinition() {
+        super(PATH,
+                DomainManagementResolver.getResolver(NotificationRegistrarsRootResourceDefinition.RESOLVER_KEY, RESOLVER_KEY),
+                NotificationRegistrarAddHandler.INSTANCE,
+                NotificationRegistrarRemoveHandler.INSTANCE);
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        OperationStepHandler writeAttributeHandler = new NotificationRegistrarWriteAttributeHandler(PATH.getKey(), ATTRIBUTES);
+        for (AttributeDefinition attr : ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(attr, null, writeAttributeHandler);
+        }
+    }
+
+    static ServiceName getServiceName(PathAddress address) {
+        return SERVICE_NAME_BASE.append(address.getLastElement().getValue());
+    }
+
+    static void installService(OperationContext context, PathAddress pathAddress, ModelNode model) throws OperationFailedException {
+        final String className = CODE.resolveModelAttribute(context, model).asString();
+        final ModuleIdentifier module = ModuleIdentifier.fromString(MODULE.resolveModelAttribute(context, model).asString());
+        final ServiceName serviceName = getServiceName(pathAddress);
+
+        final ModelNode propsNode = model.get(PROPERTIES.getName());
+        final Map<String, String> properties;
+        if (!propsNode.isDefined()) {
+            properties = Collections.emptyMap();
+        } else {
+            properties = new HashMap<>();
+            for (String key : propsNode.keys()) {
+                String value = propsNode.hasDefined(key) ? propsNode.get(key).asString() : null;
+                properties.put(key, value);
+            }
+        }
```

Comment:
Once I fix https://issues.jboss.org/browse/WFCORE-1557 in a minute you can use Properties.unwrap(context, model) to replace L109-119.